app/pages/lab/project.cjsx

```diff
@@ -232,12 +235,12 @@ EditProjectPage = React.createClass
 
 module.exports = React.createClass
   displayName: 'EditProjectPageWrapper'
-  mixins: [TitleMixin, History]
+  mixins: [TitleMixin]
   title: 'Edit'
 
   componentWillReceiveProps: (nextProps) ->
     unless nextProps.user
-      @history.pushState(null, "/lab")
+      @context.router.push '/lab'
```

Comment:
It wasn't :( I've added that in, I thought about passing the router down as a prop to the nested component, but I'm going to get it from `context` as well for consistency.